torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -110,7 +110,7 @@ void RRefContext::handleException(const JitFuture& jitFuture) {
   if (jitFuture.hasError()) {
     auto errMsg = jitFuture.tryRetrieveErrorMessage();
     VLOG(1) << "Got exception: " << errMsg;
-    TORCH_CHECK(false, errMsg);
+    TORCH_CHECK_MSG(false, errMsg);
```

Comment:
I don't see how this change is supposed to fix the bug discussed in https://github.com/pytorch/pytorch/issues/71458. In particular it seems like we should be addressing the callers of `handleException` or refactor this code not to throw.

cc @kumpera in case I'm missing something.

Reply:
@kumpera @rohan-varma 
My understanding is TORCH_CHECK will throw an new exception, but I found that there is no "try catch" when caller invoke the handleException, I think the issue is that we must handle this kind of situation.
There are two ways:
 1. handle the exception which throw by TORCH_CHECK in caller.
 2. replace TORCH_CHECK by TORCH_CHECK_MSG which doesn't throw exception just print message.

I think the method #2 is better because usually we should not throw exception again in handleException

Thanks.